jobs/Frontend/movieDatabase/src/views/main.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {FC, useEffect, useState} from 'react';
+import 'purecss/build/pure.css'
+
+import {getMovieById, getMoviesByTitle} from "../utils/moviesService";
+import {MoviesList} from "../components/moviesList";
+import './main.less'
+import {MovieDetails, MovieListItem} from "../types/movies";
+import {MovieDetail} from "../components/movieDetail";
+
+export const MainView: FC = () => {
+    const [movies, setMovies] = useState<MovieListItem[]>()
+    const [selectedMovieId, setSelectedMovieId] = useState<number>()
+    const [selectedMovieDetails, setSelectedMovieDetails] = useState<MovieDetails>()
+    const [searchTitle, setSearchTitle] = useState<string>("")
+    const [sortBy, setSortBy] = useState<string>('Name')
+    const [isLoadingList, setIsLoadingList] = useState<boolean>(false)
+    const [isLoadingDetail, setIsLoadingDetail] = useState<boolean>(false)
+
+    useEffect(() => {
+            const validationTimer = setTimeout(() => {onUserStopTyping(); }, 700)
+            return () => {
+                clearTimeout(validationTimer);
+            };
+        },
+        [searchTitle]
+    );
+
+    useEffect(() => {
+        if (movies) {
+            const result = [...movies.sort(movieSort)]
+            setMovies(result);
+            }
+        },
+        [sortBy]
+    );
+
+    useEffect(() => {
+        const getMovieDetailData = async () => {
+            setIsLoadingDetail(true)
+            const result = await getMovieById(selectedMovieId)
+            setSelectedMovieDetails(result)
+            setIsLoadingDetail(false)
+        }
+
+        if (selectedMovieId) {
+            getMovieDetailData()
+        }
+
+        },
+        [selectedMovieId]
+    );
+
+    const movieSort = (a: MovieListItem, b: MovieListItem) => {
+        switch (sortBy) {
+            case 'Name':
+                return a.title.toLowerCase().localeCompare(b.title.toLowerCase())
+                break;
+            case 'Release Date':
+                return new Date(b.release_date).getTime() - new Date(a.release_date).getTime()
+                break;
+            case 'Vote Average':
+                return b.vote_average - a.vote_average
+            break
+        }
+    }
+
+    const onUserStopTyping = async () => {
+        if (searchTitle) {
+            setIsLoadingList(true)
+            const result = await getMoviesByTitle(searchTitle);
+            result.sort(movieSort)
+            setMovies(result);
+            setIsLoadingList(false)
+        }
+    }
+
+    const getLoadingSpinner = () => {
+        return <div className={"loadingContainer"}>
+            <div id={"loading"}/>
+        </div>
+    }
+
+    return (
+        <div>
+            <div className="pure-g movieSearch">
+                <div className="pure-u-1-3"/>
+                <div className="pure-u-1-3">
+                    <h1>&#127909; Search Movie Titles</h1>

```

Comment:
🔨 All this could be moved into separated Header component which would made this component clean